PR: https://github.com/odoo/odoo/pull/277685

From: eb27c49b50611d19e30b56a27c9675d28b4cd7c7
From: paan-odoo
Date: 2026-08-25 09:01:58

Breaking data model changes scores: del:3 + add:0, change matches:
-    order_edit_tracking = fields.Boolean(string="Track orders edits", help="Store edited orders in the backend", default=False)
-    order_edit_tracking = fields.Boolean(related="config_id.order_edit_tracking", readonly=True)
-    pos_order_edit_tracking = fields.Boolean(related="pos_config_id.order_edit_tracking", readonly=False)

Total Changes: 31

[IMP] pos_*: always track edited orders

*: point_of_sale, pos_hr_restaurant, pos_restaurant, pos_self_order

Before this commit:

* Tracking edited PoS orders in the chatter required enabling a dedicated
  configuration option.

After this commit:

* Edited PoS orders are always tracked in the chatter, removing the need for any
  additional configuration.

Related PR:
* Upgrade: https://github.com/odoo/upgrade/pull/10837

Task-6400946

closes odoo/odoo#277685

Related: odoo/upgrade#10837
Signed-off-by: Loukas Wets (lowe) <lowe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -197,7 +197,6 @@ class PosConfig(models.Model):
     use_closing_entry_by_product = fields.Boolean(
         string='Closing Entry by product',
         help="Display the breakdown of sales lines by product in the automatically generated closing entry.")
-    order_edit_tracking = fields.Boolean(string="Track orders edits", help="Store edited orders in the backend", default=False)
     last_data_change = fields.Datetime(string='Last Write Date', readonly=True, compute='_compute_local_data_integrity', store=True)
     fallback_nomenclature_id = fields.Many2one('barcode.nomenclature', string="Fallback Nomenclature")
     use_custom_receipt_info = fields.Boolean(string="Customise info", default=False, help="Fill in if your shop does not have the same info as your company")

--- a/addons/point_of_sale/models/pos_order.py
+++ b/addons/point_of_sale/models/pos_order.py
@@ -309,7 +309,6 @@ class PosOrder(models.Model):
     mobile = fields.Char(string='Mobile', compute="_compute_contact_details", readonly=False, store=True)
     is_edited = fields.Boolean(string='Edited', compute='_compute_is_edited')
     has_deleted_line = fields.Boolean(string='Has Deleted Line')
-    order_edit_tracking = fields.Boolean(related="config_id.order_edit_tracking", readonly=True)
     available_payment_method_ids = fields.Many2many('pos.payment.method', related='config_id.payment_method_ids', string='Available Payment Methods', readonly=True, store=False)
     invoice_status = fields.Selection([
         ('invoiced', 'Fully Invoiced'),

--- a/addons/point_of_sale/models/pos_order_line.py
+++ b/addons/point_of_sale/models/pos_order_line.py
@@ -122,10 +122,10 @@ class PosOrderLine(models.Model):
         return super().create(vals_list)
 
     def write(self, vals):
-        if self.order_id.config_id.order_edit_tracking and vals.get('qty') is not None and vals.get('qty') < self.qty:
-            self.is_edited = True
+        if vals.get('qty') is not None and vals.get('qty') < self.qty:
+            vals['is_edited'] = True
             body = _("%(product_name)s: Ordered quantity: %(old_qty)s", product_name=self.full_product_name, old_qty=self.qty)
-            body += Markup("&rarr;") + str(vals.get('qty'))
+            body += Markup(" &rarr; ") + str(vals.get('qty'))
             for line in self:
                 line.order_id.message_post(body=line.order_id._prepare_pos_log(body))
         return super().write(vals)
@@ -134,7 +134,7 @@ class PosOrderLine(models.Model):
     def _unlink_except_order_state(self):
         if self.filtered(lambda x: x.order_id.state not in ["draft", "cancel"]):
             raise UserError(_("You can only unlink PoS order lines that are related to orders in new or cancelled state."))
-        for line in self.filtered(lambda line: line.order_id.config_id.order_edit_tracking):
+        for line in self:
             line.order_id.has_deleted_line = True
             body = _("%(product_name)s: Deleted line (quantity: %(qty)s)", product_name=line.full_product_name, qty=line.qty)
             line.order_id.message_post(body=line.order_id._prepare_pos_log(body))

--- a/addons/point_of_sale/models/pos_session.py
+++ b/addons/point_of_sale/models/pos_session.py
@@ -480,17 +480,15 @@ class PosSession(models.Model):
         if statement:
             statement._compute_balance_end_real()
 
-        if self.config_id.order_edit_tracking:
-            edited_orders = self.get_session_orders().filtered(lambda o: o.is_edited)
-            if len(edited_orders) > 0:
-                order_links = Markup().join(
-                    Markup("<li>%s</li>") % order._get_html_link() for order in edited_orders
-                )
-                body = _(
-                    "Edited order(s) during the session:%s",
-                    Markup("<br/><ul>%s</ul>") % order_links,
-                )
-                self.message_post(body=body)
+        if edited_orders := self.get_session_orders().filtered(lambda o: o.is_edited):
+            order_links = Markup().join(
+                Markup("<li>%s</li>") % order._get_html_link() for order in edited_orders
+            )
+            body = _(
+                "Edited order(s) during the session:%s",
+                Markup("<br/><ul>%s</ul>") % order_links,
+            )
+            self.message_post(body=body)
 
         if self.env.user.email:
             self.post_close_register_message()

--- a/addons/point_of_sale/models/res_config_settings.py
+++ b/addons/point_of_sale/models/res_config_settings.py
@@ -119,7 +119,6 @@ class ResConfigSettings(models.TransientModel):
     pos_note_ids = fields.Many2many(related='pos_config_id.note_ids', readonly=False)
     pos_module_pos_sms = fields.Boolean(related="pos_config_id.module_pos_sms", readonly=False)
     pos_use_closing_entry_by_product = fields.Boolean(related='pos_config_id.use_closing_entry_by_product', readonly=False)
-    pos_order_edit_tracking = fields.Boolean(related="pos_config_id.order_edit_tracking", readonly=False)
     pos_basic_receipt = fields.Boolean(related='pos_config_id.basic_receipt', readonly=False)
     pos_fallback_nomenclature_id = fields.Many2one(related='pos_config_id.fallback_nomenclature_id', domain="[('id', '!=', barcode_nomenclature_id)]", readonly=False)
     group_pos_preset = fields.Boolean(string="Presets", implied_group="point_of_sale.group_pos_preset", help="Hide or show the Presets menu in the Point of Sale configuration.")
